cases/admin.py

```
@@ -493,6 +497,55 @@ def state_badge(self, obj):
+            # Disabled button with tooltip
+            view_button = (
+                "<span "
+                'class="button" '
+                'title="Only published cases can be viewed publicly" '
+                'style="padding: 5px 10px; text-decoration: none; '
+                "background-color: #cccccc; color: #666666; border-radius: 4px; "
+                'display: inline-block; font-size: 13px; cursor: not-allowed;">'
+                "View on Site</span>"
+            )
+
+        return mark_safe(edit_button + view_button)
```

Ruff (0.15.9)

[error] Use of `mark_safe` may expose cross-site scripting vulnerabilities

(S308)